### PR TITLE
Update base image to node:20.10-buster-slim

### PR DESCRIPTION
**Pull request by AllanGuigou**
Add dependency on @octokit/rest to satisfy build requirements. Without @octokit/rest we would need to install `@octokit/core`, `@octokit/plugin-paginate-rest` and `@octokit/plugin-rest-endpoint-methods`.

This should resolve https://github.com/actions/first-interaction/issues/286

### CODE REVIEW

**Comment by pje** [comment]
Seems like we have a build issue after the node upgrade: 

```
internal/modules/cjs/loader.js:1032
  throw err;
  ^

Error: Cannot find module 'node:assert'
Require stack:
- /node_modules/undici/lib/handler/RetryHandler.js
- /node_modules/undici/index.js
- /node_modules/@actions/http-client/lib/index.js
- /node_modules/@actions/core/lib/oidc-utils.js
- /node_modules/@actions/core/lib/core.js
- /lib/main.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1029:15)
    at Function.Module._load (internal/modules/cjs/loader.js:898:27)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
    at Object.<anonymous> (/node_modules/undici/lib/handler/RetryHandler.js:1:[16](https://github.com/actions/first-interaction/actions/runs/7024356281/job/19112904850?pr=287#step:4:17))
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:[19](https://github.com/actions/first-interaction/actions/runs/7024356281/job/19112904850?pr=287#step:4:20)) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/node_modules/undici/lib/handler/RetryHandler.js',
    '/node_modules/undici/index.js',
    '/node_modules/@actions/http-client/lib/index.js',
    '/node_modules/@actions/core/lib/oidc-utils.js',
    '/node_modules/@actions/core/lib/core.js',
    '/lib/main.js'
  ]
}
```

Otherwise this makes sense and I'm happy to 👍 when the build's passing!

**Comment by AllanGuigou** [comment]
> Seems like we have a build issue after the node upgrade:
> 
> ```
> internal/modules/cjs/loader.js:1032
>   throw err;
>   ^
> 
> Error: Cannot find module 'node:assert'
> Require stack:
> - /node_modules/undici/lib/handler/RetryHandler.js
> - /node_modules/undici/index.js
> - /node_modules/@actions/http-client/lib/index.js
> - /node_modules/@actions/core/lib/oidc-utils.js
> - /node_modules/@actions/core/lib/core.js
> - /lib/main.js
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1029:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:898:27)
>     at Module.require (internal/modules/cjs/loader.js:1089:19)
>     at require (internal/modules/cjs/helpers.js:73:18)
>     at Object.<anonymous> (/node_modules/undici/lib/handler/RetryHandler.js:1:[16](https://github.com/actions/first-interaction/actions/runs/7024356281/job/19112904850?pr=287#step:4:17))
>     at Module._compile (internal/modules/cjs/loader.js:1200:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
>     at Module.load (internal/modules/cjs/loader.js:1049:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:937:14)
>     at Module.require (internal/modules/cjs/loader.js:1089:[19](https://github.com/actions/first-interaction/actions/runs/7024356281/job/19112904850?pr=287#step:4:20)) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/node_modules/undici/lib/handler/RetryHandler.js',
>     '/node_modules/undici/index.js',
>     '/node_modules/@actions/http-client/lib/index.js',
>     '/node_modules/@actions/core/lib/oidc-utils.js',
>     '/node_modules/@actions/core/lib/core.js',
>     '/lib/main.js'
>   ]
> }
> ```
> 
> Otherwise this makes sense and I'm happy to 👍 when the build's passing!

@pje was that from https://github.com/actions/first-interaction/actions/runs/7024356281? I believe that is the current first-interaction action version 😄

File: package.json [hunk 1]
```diff
@@ -29,8 +29,10 @@
     "@actions/core": "file:toolkit/actions-core-1.10.0.tgz",
     "@actions/exec": "file:toolkit/actions-exec-1.1.1.tgz",
     "@actions/github": "file:toolkit/actions-github-5.1.1.tgz",
+    "@actions/http-client": "^2.2.0",
```
[the remark this answers is not in the document]
Comment (AllanGuigou) [reply package.json:32]: also had to update the license manually from `other` to `mit` -> .licenses/npm/@actions/http-client.dep.yml

[the remark this answers is not in the document]
Comment (rentziass) [reply package.json:32]: The license issue is due to http-client [`LICENSE`](https://github.com/actions/toolkit/blob/main/packages/http-client/LICENSE) being oddly formatted, we should probably fix that.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:14.4.0-buster-slim
+FROM node:18.18.2-buster-slim
```
Comment (rentziass) [inline Dockerfile:1]: We're currently updating all first-party actions to node20, do you think we should do that here too?

**Review by rentziass — APPROVED** [review]
LGTM :shipit:

**Comment by tuhaihe** [comment]
It works, thanks!

**Review by Shepard777 — COMMENTED** [review]
Hi

**Review by Letsgoque — APPROVED** [review]

**Review by max09000-tec — APPROVED** [review]

**Pull request merged**